astropy/units/format/generic.py

```diff
@@ -592,6 +600,41 @@ def _do_parse(cls, s, debug=False):
                 else:
                     raise ValueError(f"Syntax error parsing unit '{s}'")
 
+    @classmethod
+    def _validate_unit(
+        cls, unit: str, detailed_exception: bool = True, class_name: str | None = None
```

Comment:
How about instead of adding a new parameter, we allow classes to define their own name? A bit like we do with `baseframe`. And then do something like `class_name = getattr(cls, "name", cls.__name__)`. Or even go the full hog and do:
```
def __init_subclass__(cls, **kwargs):
    if not hasattr(cls, "name"):
        cls.name = cls.__name__
```

Reply:
The formatter classes already have a `name`, which is publicly used by the formatter registry: https://github.com/astropy/astropy/blob/3be7d5631a03c35811f0cfd16b3c41f670aa7724/astropy/units/format/base.py#L32-L38
It is the lowercase version of `cls.__name__`,  except for `LatexInline` which has an explicitly specified name for better readability: https://github.com/astropy/astropy/blob/3be7d5631a03c35811f0cfd16b3c41f670aa7724/astropy/units/format/latex.py#L83
I would find a second name attribute confusing.

Reply:
Ah, I was wondering about a name, but had forgotten we already had it! Maybe we should just use `cls.name` in the error messages? After all, that is how they are called by the user.

Reply:
But the messages here refer to the OGIP, FITS or VOUnit standards, not to the `"ogip"`, `"fits"` or `"vounit"` formats. I would be more comfortable with renaming the `Fits` class to `FITS` so we could just use `cls.__name__` here.

Reply:
Yes, on further thought, those classes are not used at all outside of the format piece; I think it is fine to rename, especially as the all-capital version really is more logical - we do the same for OGIP.

Reply:
It would be best to open a separate pull request for renaming the `Fits` class to `FITS` because it is an API change and shouldn't be done in a refactoring pull request.

I'll leave it to you to decide if we should put this pull request on hold until `Fits` has been renamed. The alternative would be to drop the second commit here, merge the first commit regarding `_to_decomposed_alternative()` as it is and then to open a third pull request for `_validate_unit()` after the `Fits` renaming pull request has been merged. It's getting late in my timezone, so if you think merging the first commit right away is better then you can just go ahead and do it.